compiler+runtime/src/cpp/jank/runtime/obj/inst.cpp

```diff
@@ -0,0 +1,102 @@
+#include <chrono>
+#include <iostream>
+#include <locale>
+#include <sstream>
+
+#include <jank/runtime/core/make_box.hpp>
+#include <jank/runtime/obj/inst.hpp>
+#include <jank/runtime/rtti.hpp>
+#include <jank/util/fmt/print.hpp>
+
+namespace jank::runtime::obj
+{
+  static jtl::ref<inst_time_point> now()
+  {
+    return jtl::make_ref<inst_time_point>(std::chrono::system_clock::now());
+  }
+
+  inst::inst()
+    : value{ now() }
+  {
+  }
+
+  static jtl::ref<inst_time_point> inst_from_string(jtl::immutable_string const &s)
+  {
+    inst_time_point o;
+    auto const formats = { "%FT%T%Oz", "%FT%T%z", "%FT%T", "%F" };
+    auto parsed = false;
```

Comment:
Please always use direct initialization. i.e. no ` = `.

Also, `formats` can be `static`, since we don't need to allocate space for it every call.

Reply:
So for bools, direct initialization is `bool parsed{ false };`, but how do I direct initialize formats?

Reply:
Same thing for both of them. You're already using type inference, but you're putting a ` = ` which does nothing. Just remove the ` = ` from your original code and use `{}`.

```cpp
auto const formats{ "%FT%T%Oz", "%FT%T%z", "%FT%T", "%F" };'
auto parsed{ false };
```

Reply:
I think the different length strings is messing with auto:

```
inst.cpp:19:44: error: initializer for variable 'formats' with type 'const auto' contains multiple expressions
   19 |     static auto const formats{ "%FT%T%Oz", "%FT%T%z", "%FT%T", "%F" };
```
however, the following works:

```cpp
    static std::initializer_list<char const *> const formats{ "%FT%T%Oz",
                                                              "%FT%T%z",
                                                              "%FT%T",
                                                              "%F" };
```

is this suitable?

Reply:
My fault. I forgot this rule [changed](https://stackoverflow.com/questions/51166807/initializer-list-with-auto-contains-multiple-expressions) in the standard. I'd probably just do this. The compiler can figure out the template arg itself.

```cpp
static std::vector const formats{ "%FT%T%Oz", "%FT%T%z", "%FT%T", "%F" };
```